presto-main/src/main/java/com/facebook/presto/sql/planner/DependencyExtractor.java

```diff
@@ -36,8 +37,13 @@ private DependencyExtractor() {}
 
     public static Set<Symbol> extractUnique(PlanNode node)
     {
+        return extractUnique(node, n -> n);
+    }
+
+    public static Set<Symbol> extractUnique(PlanNode node, Lookup lookup)
```

Comment:
Before this change, a `Rule` was able to extract symbols from a `PlanNode` without recursing into sources. This property turned out to be important (https://github.com/prestodb/presto/pull/7888/files#diff-1a430cab232220a3d693ed1d27066e9fR377), so could you retain `extractUnique` overload that does that?